src/modules/fw_pos_control_l1/fw_pos_control_l1_main.cpp

```diff
@@ -2299,10 +2299,22 @@ FixedwingPositionControl::task_main()
 				// adjust navigation waypoints in position control mode
 				if (_control_mode.flag_control_altitude_enabled && _control_mode.flag_control_velocity_enabled
 				    && _global_pos.lat_lon_reset_counter != _pos_reset_counter) {
+
+					// add position reset delta to previous waypoint coordinate
 					_hdg_hold_prev_wp.lat += _global_pos.delta_lat_lon[0];
+					_hdg_hold_prev_wp.lat = M_RAD_TO_DEG * asin(sin(_hdg_hold_prev_wp.lat * M_DEG_TO_RAD));
```

Comment:
This seems like a costly operation, however, it's only used in case of a reset, so that should not be a problem.

But, please add a comment (and maybe even link to the stackoverflow page if you found it there). A future reader will be puzzled about what the hell this does :smile:.